### PR TITLE
Add Config struct to rtsp module

### PR DESCRIPTION
**Pull request by randhid**
We're removing builtin/rtsp, but this module has a dependency on that package. The only thing it's doing is pulling the Config Struct and Validate function from that package. I move them here.

### CODE REVIEW

File: cmd/remote/cmd.go [hunk 1]
```diff
@@ -26,7 +25,7 @@ func main() {
 func realMain() error {
 
 	ctx := context.Background()
-	logger := golog.NewDevelopmentLogger("client")
+	logger := logging.NewLogger("client")
```
Comment (randhid) [inline cmd/remote/cmd.go:28]: The module would not build locally unless I changed the logger to the new  style logger in the rdk. runWebWithConfig needed a `logging.Logger` for me. Can rever if this is not the case on whatever this is being built on.

File: rtsp.go [hunk 2]
```diff
@@ -21,19 +21,19 @@ import (
 	goutils "go.viam.com/utils"
 
 	"go.viam.com/rdk/components/camera"
-	"go.viam.com/rdk/components/camera/rtsp"
```
Comment (randhid) [inline rtsp.go:24]: This package will eventually go away. Config used to be : https://github.com/viamrobotics/rdk/blob/0a7f6b23851f8c04f82ffa5583f600668e5e2e3a/components/camera/rtsp/rtsp.go#L46

**Comment by randhid** [comment]
@erh - this removed the dependency of this module on the builtin rtsp camera go package.

cc @seanavery @nicksanford

**Comment by erh** [comment]
can you also upgrade to newest rdk as part of this.

File: cmd/remote/cmd.go (hunk 1 above)
Comment (nicksanford) [reply cmd/remote/cmd.go:28]: Changing to `logging.NewLogger` will no longer log debug-level logs (which the previous logger did).

From a cursory glance, the only code in this repo that logs at debug level is here: https://github.com/erh/viamrtsp/blob/main/rtsp.go#L128

Could I recommend changing this to:
```suggestion
	logger := module.NewLoggerFromArgs("client")
```

That way module will default to info level logs but allow the end user to configure the level at runtime: https://pkg.go.dev/go.viam.com/rdk@v0.23.2/module#NewLoggerFromArgs

**Review by nicksanford — APPROVED** [review]
Looks good minus nit & updating to latest RDK

File: cmd/remote/cmd.go (hunk 1 above)
Comment (randhid) [reply cmd/remote/cmd.go:28]: Thanks! I'll make the change.

**Comment by nicksanford** [comment]
@randhid do you not also need to remove this line from cmd/remote/main.go: https://github.com/erh/viamrtsp/pull/4/files#diff-788d8d6232a7bfca6baf42aeb2f5334425bade25a2d8ee4ca3ba6e89cb2fc7afR8?

**Comment by randhid** [comment]
> @randhid do you not also need to remove this line from cmd/remote/main.go: https://github.com/erh/viamrtsp/pull/4/files#diff-788d8d6232a7bfca6baf42aeb2f5334425bade25a2d8ee4ca3ba6e89cb2fc7afR8?

yep, thanks for catching that.

File: .canon.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+rtsp:
```
Comment (randhid) [inline .canon.yaml:1]: @nicksanford is this useful for you and sean or should I add it to the git ignore?

Comment (nicksanford) [reply .canon.yaml:1]: I don't mind it :) 

It can stay from my perspective

File: .canon.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    image_amd64: ghcr.io/viamrobotics/canon:amd64
+    image_arm64: ghcr.io/viamrobotics/canon:arm64
+    minimum_date: 2022-12-20T18:49:25.781750658Z
+    update_interval: 168h0m0s
+    persistent: false
+    user: testbot
+    group: testbot
```
Comment (randhid) [inline .canon.yaml:8]:
```suggestion
    group: testbot
    
```

File: .canon.yaml (hunk 3 above)
Comment (seanavery) [reply .canon.yaml:1]: Canon wrapper is useful - using `antique` containers for static build PR.

**Review by seanavery — COMMENTED** [review]
LGTM!
- Will rebase [Codec Detection & H265 PR](https://github.com/erh/viamrtsp/pull/2) with this change.

File: .canon.yaml (hunk 3 above)
Comment (randhid) [reply .canon.yaml:1]: I'll add that!

File: .gitignore [hunk 5]
```diff
@@ -26,3 +26,4 @@ go.work
 
 viamrtsp
 module.tar.gz
+
```
Comment (erh) [inline .gitignore:29]: remove

**Review by erh — APPROVED** [review]
minus that one nit

File: .gitignore (hunk 5 above)
Comment (randhid) [reply .gitignore]: removed.

**Pull request merged**